### PR TITLE
Add options to conditionalize building the docs

### PR DESCRIPTION
**Pull request by bstinsonmhk**
We want to do this in case the OS does not ship pandoc.

Signed-off-by: Brian Stinson <bstinson@redhat.com>

### CODE REVIEW

File: redhat/rdma-core.spec [hunk 1]
```diff
@@ -301,6 +315,9 @@ easy, object-oriented access to IB verbs.
          -DCMAKE_INSTALL_UDEV_RULESDIR:PATH=%{_udevrulesdir} \
          -DCMAKE_INSTALL_PERLDIR:PATH=%{perl_vendorlib} \
          -DENABLE_IBDIAGS_COMPAT:BOOL=True \
+%if ! %{with builddocs} 
+         -DNO_MAN_PAGES=1 \
```
Comment (jgunthorpe) [inline redhat/rdma-core.spec:319]: No, package build scripts can never include NO_MAN_PAGES - shipping rdma-core without man pages is completely unacceptable.

**Review by jgunthorpe — CHANGES_REQUESTED** [review]

File: redhat/rdma-core.spec (hunk 1 above)
Comment (bstinsonmhk) [reply redhat/rdma-core.spec:319]: What if we switched the bcondition at the top to always build docs by default? The problem I'm trying to solve here is to not require pandoc in downstream specs (which are already shipping prebuilt docs anyways), while allowing us to do test builds against the upstream spec too.

Comment (jgunthorpe) [reply redhat/rdma-core.spec:319]: The cmake automatically uses the prebuilt docs, there is no need to change the specfile at all.

Comment (bstinsonmhk) [reply redhat/rdma-core.spec:319]: I'm trying to remove the BuildRequires on pandoc from the spec. So to ask my question another way: is there some way I can use this upstream spec to build a non-production build on an OS that does not ship pandoc?

Comment (jgunthorpe) [reply redhat/rdma-core.spec:319]: Remove the buildrequires and everything will work if your source tar has the correct pre-built man pages in it.

Comment (Honggang-LI) [reply redhat/rdma-core.spec:319]: > Remove the buildrequires and everything will work if your source tar has the correct pre-built man pages in it.

**_NO_**, this won't work for source tarball generated from upstream git repo. We **always** have to test patch against latest
upstream repo.

Comment (nmorey) [reply redhat/rdma-core.spec:319]: For SUSE we use the git repo but pre-build the doc before submitting to our build system.
It adds a step but solved the pandoc dependency issue for us.

Comment (jgunthorpe) [reply redhat/rdma-core.spec:319]: @Honggang-LI if it doesn't work then the source tgz you are using doesn't have docs prebuilt and, again, complete NO to distros shipping without docs. You have to use tgz with prebuilt doc, and if you do then no changes to the spec file are required. See Nichola's note for how suse does it.

Comment (nmorey) [reply redhat/rdma-core.spec:319]: @Honggang-LI The spec file and the gen-pandoc.sh script can be found here if you want to have a look:
https://build.opensuse.org/package/show/science:HPC/rdma-core

**Comment by Honggang-LI** [comment]
@rleon 

We need a script to build a cannonical distribution tar from pure source. As the PR is closed now, is the script available for us?

**Comment by nmorey** [comment]
@Honggang-LI The release script from AZP is available in `buildlib/azure-pipelines-release.yml`

**Comment by rleon** [comment]
I closed it because this PR is not going to be merged.

**Comment by Honggang-LI** [comment]
> @Honggang-LI The release script from AZP is available in `buildlib/azure-pipelines-release.yml`

Well, I don't have an Azure account to run this pipeline script. We need a docker script.

**Comment by nmorey** [comment]
@Honggang-LI Can't you just extract the script part and package it however you need it ?
```
$ python3 -c "import yaml; print(yaml.safe_load(open('buildlib/azure-pipelines-release.yml'))['stages'][0]['jobs'][0]['steps'][1]['bash'])"
set -e
mkdir build-pandoc artifacts
cd build-pandoc
CC=gcc-10 cmake -GNinja ..
ninja docs

cd ..
python3 buildlib/cbuild make-dist-tar build-pandoc

```

**Comment by Honggang-LI** [comment]
> @Honggang-LI Can't you just extract the script part and package it however you need it ?

We need docker container script. Your suggestion will not work for RHEL9 users, as pandoc will not available for RHEL9.

**Comment by nmorey** [comment]
@Honggang-LI I get that.
But can't you use something like the script above to build a "release" tarball from any commit in the git (in a CI tool for example, on any distro that has pandoc). And then use this tarball to run whatever you need to on a RHEL9.

**Pull request closed without merging**